listenbrainz_spark/recommendations/candidate_sets.py

```diff
@@ -81,17 +81,22 @@ def get_listens_to_fetch_top_artists(mapped_df, from_date, to_date):
     return mapped_listens_subset
 
 
-def get_top_artists(mapped_listens_subset, top_artist_limit):
+def get_top_artists(mapped_listens_subset, top_artist_limit, users):
     """ Get top artists listened to by users who have a listening history in
         the past X days where X = RECOMMENDATION_GENERATION_WINDOW.
 
         Args:
             df (dataframe): A subset of mapped_df containing user history.
             top_artist_limit (int): number of top artist to calculate
+            users: list of users to generate candidate sets.
 
         Returns:
-            top_artists_df (dataframe): Top Y artists listened to by a user for all users where
-                                        Y = TOP_ARTISTS_LIMIT
+            if users is an empty list:
+                top_artists_df (dataframe): Top Y artists listened to by a user for all users where
+                                            Y = TOP_ARTISTS_LIMIT
+            else:
+                top_artist_given_users_df (dataframe): Top Y artists listened to by a user for given users where
+                                                       Y = TOP_ARTISTS_LIMIT
     """
     df = mapped_listens_subset.select('mb_artist_credit_id', 'msb_artist_credit_name_matchable', 'user_name') \
```

Comment:
`df` is being used to generate `top_artist_df`. If `users` is an empty list, `top_artist_df` will be returned as such otherwise top artist for given users will be filtered from from `top_artist_df`.